### ISSUE TITLE
Deeply nested IndexLink at / isn't active

### ISSUE THREAD

**Issue by CrisLi** [rackt/react-router#2212]
My router configuration is like below.

```jsx
  <Router history={history}>
    <Route component={App}>
      <Route path="/login" component={Login} />
      <Route path="/" onEnter={requireAuth} component={Home}>
        <IndexRoute component={Dashboard} />
        <Route path="/about" component={About} />
      </Route>
      <Route path="*" component={NotFound} />
    </Route>
  </Router>

```

```jsx
    return (
            <IndexLink
              to="/"
              activeClassName="active">Dashboard
            </IndexLink>
            <Link
              to="/about"
              activeClassName="active">About
            </Link>
    );
```

In 1.0.0-rc1, I only need to use ```Link``` to active each <a> tag. But in the 1.0.0-rc3, I have to use ```IndexLink``` component. But the ```IndexLink``` is not active unless I change the router config lilke:

```jsx
  <Router history={history}>    
      <Route path="/login" component={Login} />
      <Route path="/" onEnter={requireAuth} component={Home}>
        <IndexRoute component={Dashboard} />
        <Route path="/about" component={About} />
      </Route>
      <Route path="*" component={NotFound} />
  </Router>

```

It seems like the ```InexLink``` only work for the root index. How to config a sub-index?

**Comment by mjackson** [comment]
I'm sorry but I'm unable to reproduce the bug you're describing. In d9053829722509dae07826fb86b5e51918963aab I verified that `IndexLink` is working correctly on deeply nested `IndexRoute`s. Can you take a look at that example? If there's something you're doing that we're not doing there, I'm happy to re-open and keep discussing.

**Comment by CrisLi** [comment]
I have upload a demo project to https://github.com/CrisLi/react-router-demo

Please have look at it. When I click the 'Dashboard' link, it would not become 'active'. Let know if there anything I was wrong. Thanks a lot.

**Comment by jepezi** [comment]
@CrisLi @mjackson Have you resolved this issue? I have the same issue here.
I've also created very simple repo here https://github.com/jepezi/react-router-rc3-activeclass

**Comment by CrisLi** [comment]
@jepezi Not yet.
@mjackson I think maybe we should re-open this issue. It seems like this is a defect.

**Comment by jepezi** [comment]
Investigating further. I've found that when the route's path is NOT '/', then IndexRoute is active but not when path is '/'. Consider this route and nav.

```js
const routes =
{component: App, childRoutes: [
  // path: '/website' here causes Home route active
  {path: '/website', component: Wrapper, indexRoute: {component: Home}, childRoutes: [
    {path: 'product_category', component: ProductCategory},
  ]},
  {path: '/login', component: Login},
  {path: '*', component: NotFound},
]};

// IndexLink here is active
<ul>
  <li><IndexLink to="/website" activeClassName="active">Home</IndexLink></li>
  <li><Link to="/website/product_category" activeClassName="active">Category</Link></li>
  <li><Link to="/login" activeClassName="active">login</Link></li>
</ul>
```

When we change `path: '/website' ...` to `path: '/' ...`, then Home route is not.
```js
const routes =
{component: App, childRoutes: [
  {path: '/',                //  <---------------- change path to '/'
...

// With this new link (/website removed), IndexLink here is not active.
<ul>
  <li><IndexLink to="/" activeClassName="active">Home</IndexLink></li>
  <li><Link to="/product_category" activeClassName="active">Category</Link></li>
...
```

**Comment by jepezi** [comment]
Here is the simple repo to illustrate the issue. https://github.com/jepezi/indexlink

**Comment by mjackson** [comment]
There seems to be some strange behavior at `/`. We should have a test for a deeply nested `IndexRoute` that renders at `/`.

**Comment by taion** [comment]
This was fixed by #2237.
